### ISSUE TITLE
Phylanx Python variable definition create copies of containers instead of references

### ISSUE THREAD

**Issue by parsa** [STEllAR-GROUP/phylanx#376]
This example shows that upon the definition of a variable, Phylanx create a copy of a list instead of referencing it. This behavior is different than Python's.

### Regular Python Code
```python
a = []
b = a
b += [42]
print(a, b)
```
Output:
```
[42, 42] [42, 42]
```

### PhySL code
```python
@Phylanx()
def fx1():
    a = [42]
    b = a
    b += a
    print(a, b)
fx1()
```
Output:
```
[42] [84]
<nil>
```

**Comment by hkaiser** [comment]
I don't think that we can change the default behavior of PhySL. Even more as Python is not very consistent with regards to its copy semantics. Some Python types are deeply copied, some are copied by reference (shallow copies).

**Comment by parsa** [comment]
I don't know if we should change the default behavior of PhySL but the Python code's behavior should be identical regardless of how it's actually executed.

> Python is not very consistent with regards to its copy semantics
Yes, but I'm not clear what you're getting at with this. All mutable object in Python are always references, immutable objects are always values.

**Comment by hkaiser** [comment]
> It's not clear what you're getting at. All mutable object in Python are always references, immutable objects are always values.

@parsa: integers are mutable and are not handled as references, for instance:
```
>>> a = 1    
>>> b = a    
>>> a = 2    
>>> b        
1            
```

**Comment by parsa** [comment]
Python Integers are simple types and are immutable.

**Comment by hkaiser** [comment]
@parsa I was able to change the value of the integer `a`, wasn't I?

**Comment by parsa** [comment]
@hkaiser from [The Python Language Reference](https://docs.python.org/3/reference/datamodel.html):
> An object’s mutability is determined by its type; for instance, numbers, strings and tuples are immutable, while dictionaries and lists are mutable.

**Comment by stevenrbrandt** [comment]
This ticket has a few issues. One of which is how the addition of arrays works. The change below will allow [42]+[42] to be [42,42].
```
diff --git a/src/execution_tree/primitives/base_primitive.cpp b/src/execution_tree/primitives/base_primitive.cpp
index 903d388..bcf80fc 100644
--- a/src/execution_tree/primitives/base_primitive.cpp
+++ b/src/execution_tree/primitives/base_primitive.cpp
@@ -677,6 +677,8 @@ namespace phylanx { namespace execution_tree
 
         case 5: HPX_FALLTHROUGH;    // primitive
         case 7: HPX_FALLTHROUGH;    // phylanx::ir::range
+            return std::move(val);
+
         default:
             break;
         }
```

**Comment by hkaiser** [comment]
@stevenrbrandt should arrays be supported by `+` ? Currently, `+` does support lists (i.e. `make_list(42) + make_list(42) == make_list(42, 42)`, is that what you need?

**Comment by stevenrbrandt** [comment]
Maybe I'm a commit or two behind, but I wasn't getting that behavior. I was getting the error thrown right after that segment of code I identified.
